templates/CONTRIBUTING.md

```diff
@@ -1,3 +1,57 @@
+
+## Built the project
+
+1. Clone this repo locally. (git clone https://github.com/OfficeDev/TeamsFx.git)
+1. Open a terminal and move into your local copy. (cd TeamsFx)
+1. Build the Template Builder package. (cd templates/src && npm install && npm run build)
+
+## Terminology
+
+Template: A template is the final output of Template Builder. Each template will be packed into a zip file and be released. Teams Toolkit downloads the zip file for scaffolding.
+Template Definition File: A template definition file in yaml format defines the template.
+Template Assets: A set of reusable files that compose of template.
+
+## Folder Structure
+
+* `assets`  contains the template assets that compose a template.
+* `definitions` contains the yaml files that define templates.
+* `scenarios` contains the output of template builder, each sub folder containing a complete template source code. Each template will be zipped and released by CD. A final template can be composed of several assets.
+* `src` contains the source code of template builder. Template owners can run the command in the src folder to generate their template source code in scenarios folder. The definition filename will be the template name.
+
+
+## How to add a new template?
+
+1. Add your template definition yaml file under the `definitions` folder.
+1. Add your new template assets under the `assets` folder, and refer to the assets in your template definition.
+    For example, you may need to add CSharp code for command-and-response bot. Then go to `assets/code/csharp` folder, create a new folder named `command-and-response` and put your source code in it. Finally, refer to your new assets in your template definition:
+    ```
+    assets:
+      # Application code
+      - from: code/csharp/command-and-response     # Relative path to 'assets'.
+        to: .                                      # Relative path to the destination folder. 'to' can be omitted if it is '.'.
+    ```
+1. Generate template source code with Template Builder.
+    ```
+    > cd src
+    > npm run generate ../definitions/<your-definition-file>
+    ```
+1. Commit both your definition file and your template source code.
+
+## How to synchronize template asset changes to templates?
+
+Assuming every template assets are referred by at least one template. When updating a template asset, you must synchronize the changes to those templates that refer to this asset.
+
+1. Add your changes to existing asset.
+1. You have no need to figure out which template should be updated. Re-generate all templates:
+    ```
+    > cd src
+    > npm run generate:all
+    ```
+1. Commit both your asset changes and template changes.
+
+
 ## How to scaffold from pre-release templates?
```

Comment:
Is that better for "How to test template for pre-release version"?

Reply:
Do we need case of "How to test template in local debug?"

Reply:
We have "How to debug template?" section.